databricks/spark-deep-learning
# Add documentation source, instructions for building docs to gh-pages branch

This PR:
* Updates instructions for building the spark-deep-learning docs
* Adds hosted documentation source files to the project root directory

See what the hosted docs will look like [here](https://smurching.github.io/spark-deep-learning/); if this PR is merged, the same docs will be hosted at https://databricks.github.io/spark-deep-learning.

#### Background
There are [several ways](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/) to host project documentation on Github pages; the approach used in this PR is to host docs from a `gh-pages` branch in which documentation source files are in the project root. This allows us to avoid checking in doc files to our master branch, simplifying the process of building docs during development and reducing the size of master.

#### How to test
Follow the instructions in `./docs/README.md` for building docs. A `site` folder should appear in the project root directory; open `site/index.html` to view the docs. Alternatively, use jekyll to locally serve the docs as described in the README.
